src/sqlacodegen/generators.py

```diff
@@ -674,10 +674,17 @@ def fix_column_types(self, table: Table) -> None:
 
     def get_adapted_type(self, coltype: Any) -> Any:
         compiled_type = coltype.compile(self.bind.engine.dialect)
+        adapted_type: Any = None
         for supercls in coltype.__class__.__mro__:
             if not supercls.__name__.startswith("_") and hasattr(
                 supercls, "__visit_name__"
             ):
+                if (
```

Comment:
@np-kyokyo can you, for the very least, add a comment explaining this very generic fix